app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/EligibilityController.scala

```diff
@@ -95,172 +79,104 @@ class EligibilityController @Inject()(
-    getPrevious[EligibilityJourney](store).flatMap { previous =>
+    journeyTraverseService.getPrevious[EligibilityJourney].flatMap { previous =>
       willYouClaimForm.bindFromRequest().fold(
         errors => Future.successful(BadRequest(willYouClaimPage(errors, previous))),
         form => {
-          store.update[EligibilityJourney]({ x =>
-            x.map { y =>
-              y.copy(willYouClaim = y.willYouClaim.copy(value = Some(form.value.toBoolean)))
-            }
-          }).flatMap(_.next)
+          store.update[EligibilityJourney](updateWillYouClaim(form.value.toBoolean)).flatMap(_.next)
         }
       )
     }
   }
 
   def getNotEligible: Action[AnyContent] = escAuthentication.async { implicit request =>
-    Future.successful(Ok(notEligiblePage()))
+    Ok(notEligiblePage()).toFuture
   }
 
 
   def getMainBusinessCheck: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
-    store.get[EligibilityJourney].flatMap {
-      case Some(journey) =>
-        journey
-          .mainBusinessCheck
-          .value
-          .fold(
-            Future.successful(
-              Ok(mainBusinessCheckPage(
-                mainBusinessCheckForm,
-                journey.previous
-              ))
-            )
-          ){x =>
-            Future.successful(
-              Ok(mainBusinessCheckPage(
-                mainBusinessCheckForm.fill(FormValues(x.toString)),
-                journey.previous
-              ))
-            )
-          }
-      case _ => handleMissingSessionData("Eligibility journey")
+    store.get[EligibilityJourney]
+      .map(_.getOrElse(handleMissingSessionData("Eligibility Journey")))
+      .map { journey =>
+        val form = journey.mainBusinessCheck.value.fold(mainBusinessCheckForm)(mainBC => mainBusinessCheckForm.fill(FormValues(mainBC.toString)))
+        Ok(mainBusinessCheckPage(form, journey.previous))
     }
   }
 
   def postMainBusinessCheck: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
-    getPrevious[EligibilityJourney](store).flatMap { previous =>
+    journeyTraverseService.getPrevious[EligibilityJourney]
+      .flatMap { previous =>
       mainBusinessCheckForm.bindFromRequest().fold(
         errors => Future.successful(BadRequest(mainBusinessCheckPage(errors, previous))),
         form => {
-          store.update[EligibilityJourney]({ x =>
-            x.map { y =>
-              y.copy(mainBusinessCheck = y.mainBusinessCheck.copy(value = Some(form.value.toBoolean)))
-            }
-          }).flatMap(_.next)
+          store.update[EligibilityJourney](updateMainBusinessCheck(form.value.toBoolean)).flatMap(_.next)
         }
       )
     }
   }
 
   def getNotEligibleToLead: Action[AnyContent] = escAuthentication.async { implicit request =>
-    Future.successful(Ok(notEligibleToLeadPage()))
+    Ok(notEligibleToLeadPage()).toFuture
```

Comment:
Can we keep this consistent?

Reply:
yes, we are changing it to .toFuture as we go and do changes

Reply:
Who decided that?

Reply:
I introduced some syntax a while back to clean up the `Future.successful` boilerplate in the codebase.

Reply:
carwyn started it and i thought it's not a big change and picked up. but yeah we can discuss about it 👍

Reply:
Is there an option in scalafmt to enforce suffix opitonals/futures?

I'm open to either. But the fact we're declaring futures and optionals in different ways in such a small codebase is just messy.

And FWIW - `.toFuture` and `Future.successful` are not equivalent https://blog.colinbreck.com/future-successful-blocking-your-future-success/

Reply:
@sam-miele It's not a big deal at all. The motivation here is to reduce visual noise (aka boilerplate) in the codebase so that the code is cleaner and easier to read. I've applied this pattern to many scala projects and it's always worked really well. It's not really any different to `.some` offered by cats. See also `OptionTSyntax` which I added recently to help clean up our for comprehensions.